### PR TITLE
Delete leaked objects if it's already deleted by other goroutines

### PR DESCRIPTION
**Pull request by polyrabbit**
When gc and uploading happens simultaneously on the same stage file, the uploaded object will be leaked - never got a second chance to be deleted.

We observed several cases of leaded objects when `write-back` is enabled and `trash-days` set to 0.

### CODE REVIEW

File: pkg/chunk/cached_store.go [hunk 1]
```diff
@@ -965,10 +966,19 @@ func (store *cachedStore) uploadStagingFile(key string, stagingPath string) {
...
+	if !store.isPendingValid(key) {
+		block.Release()
+		logger.Debugf("Key %s is not needed, drop it", key)
+		return
+	}
 
 	store.stageBlockDelay.Add(time.Since(item.ts).Seconds())
 	if err = store.upload(key, block, nil); err == nil {
 		store.bcache.uploaded(key, blen)
+		if !store.isPendingValid(key) { // Delete leaked objects if it's already deleted by other goroutines
```
Comment (SandyXSD) [inline pkg/chunk/cached_store.go:978]: Skip other functions (`uploaded`, `removePending`, `removeStage`) when isPendingValid = false.

Comment (polyrabbit) [reply pkg/chunk/cached_store.go:978]: Is it safe? I suppose at least we still need `removePending`?

Comment (SandyXSD) [reply pkg/chunk/cached_store.go:978]: pending not valid means it's already removed from the pending dict

Comment (polyrabbit) [reply pkg/chunk/cached_store.go:977]: Right, updated

**Pull request merged**